src/script/lua_api/l_util.cpp

```diff
@@ -320,6 +321,103 @@ int ModApiUtil::l_decompress(lua_State *L)
 	return 1;
 }
 
+#if USE_CURL
+void ModApiUtil::read_http_fetch_request(lua_State *L, HTTPFetchRequest &req)
+{
+	luaL_checktype(L, 1, LUA_TTABLE);
+
+	req.caller = httpfetch_caller_alloc();
+	getstringfield(L, 1, "url", req.url);
+	req.post_data = getstringfield_default(L, 1, "post_data", "");
+	std::string useragent = getstringfield_default(L, 1, "useragent", "");
+	lua_getfield(L, 1, "useragent");
+	bool useragent_exists = lua_isstring(L, 2);
+	lua_pop(L, 1);
+	if (useragent_exists)
+		req.useragent = getstringfield_default(L, 1, "useragent", "");
```

Comment:
Yes, but look at what your code does:
  1. Save `useragent` field or `""` into the (unused) `useragent` variable.
  2. Look up the `useragent` field (again).
  3. Check if the value is a string.
  4. Discard the. value.
  5. Look the value up yet again and this time actually store it where it's used, providing a default of `""` even though you've just made sure it exists.

Here's a sane way to do it:
```C++
lua_getfield(L, 1, "useragent");
// Note negative indexes when using pushed values.
// That way you don't have to update this if you add a second argument,
// and the function won't fail if someone passes more arguments than are used.
if (lua_isstring(L, -1))
        req.useragent = lua_tostring(L, -1);
lua_pop(L, 1);
```